### PR TITLE
refactor: move types wip

### PR DESCRIPTION
**Pull request by haxxmaxx**
- dividing types into two files: one on a nebula/data level and one on a "table" level
- about the component props: there are no perfect way of doing this. Technically those components that spread the props from TableWrapper will recieve all those props. But they only use a subset of them, and it is different for all components. We are currently using omit, which this is not really the correct use-case for (it is more for omitting and overriding props). I decided to create a common interface for all components. Turns out that it is only 3 props, but better then nothing. You could potentially extend the type gradually, but I don't think that is very future proof. And they are kind of independent, so the chaining would only be for not repeating code
- Also note that I am keeping some types in the selection-utils. I think they fit nicely there since they are very much related to the dispatch

### CODE REVIEW

File: src/table/types.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  cursor?: string;
+  borderWidth?: string;
+  backgroundColor?: string;
+  sortLabelColor?: string;
+  hoverBackgroundColor?: string;
+  hoverFontColor?: string;
+}
+
+export interface CellStyle {
+  backgroundColor: string | undefined; // This is always set but could be undefined in the theme
```
Comment (mhs30) [inline src/table/types.ts:57]: Does it makes sense of making it optional?

File: src/table/types.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  app?: EngineAPI.IApp;
+}
+
+export interface CommonTableProps {
+  tableData: TableData;
+  theme: ExtendedTheme;
+  keyboard: stardust.Keyboard;
+}
+
+export interface TableWrapperProps extends CommonTableProps {
```
Comment (mhs30) [inline src/table/types.ts:153]: Should this one and the next two extend from `TableTotalProps` aswell since they all use rootElement and layout?

**Review by mhs30 — APPROVED** [review]
Minor comment, code lgtm

File: src/table/types.ts (hunk 1 above)
Comment (haxxmaxx) [reply src/table/types.ts:57]: well I don't think so. we always set it and sometimes it undefined

File: src/table/types.ts (hunk 2 above)
Comment (haxxmaxx) [reply src/table/types.ts:153]: I thought about it, but let's say we add a prop to totals that the other three don't need, then we would have to revert it to the way it is now. We will be adding a lot features soon so these props are likely to change

Comment (mhs30) [reply src/table/types.ts:153]: Makes sense, then leave it as it is until you know more.

**Pull request merged**